Implement quote/delim encode settings

schoettl opened the pull request:
Hi, I recently needed support for some encoding settings. Are you interested to merge and publish my changes?

Basically, `quote` tells if *all* cells should be quoted and `delimiter` changes the delimiter.

If the delimiter appears in a cell value, the cell value is quoted regardless of the `quote` setting.

Many of the visible diffs are whitespace changes after elm reformatted the code.

schoettl committed:
Implement quote/delim encode settings

zenhack commented on src/Csv/Encode.elm:
```diff
@@ -41,12 +42,30 @@ type alias Csv =
 
 
-{-| A bytes encoder for `Csv`s
+type alias EncodeSettings =
```
Given that we're already in a module called `Encode`, I think we can just call this `Settings`.

schoettl replied:
Right, i changed this.

zenhack reviewed (changes requested):
Sorry for the slow response; thanks for the patch! This looks broadly reasonable, just one minor comment in-line.

schoettl committed:
Rename EncodeSettings to Settings

zenhack commented on src/Csv/Encode.elm:
```diff
@@ -41,12 +42,30 @@ type alias Csv =
+    }
+
+
+defaultEncodeSettings : Settings
```
This should also probabably just be `defaultSettings`. Otherwise LGTM.

schoettl committed:
Rename default settings

schoettl commented:
Maybe I should add some documentation to the type?
And rename `quoted` to `alwaysQuoted`? Because if `not quoted`, encode still quotes if there is a quote in the value (which is good, I think).

zenhack commented:
That would be great; if you plan on doing that I'll hold off on merging.

schoettl committed:
Add doc, fix name

schoettl committed:
Fix doc

schoettl commented:
I added doc and changed the name. I also changed my documentation because we don't want to encourage people to use a multi-character delimiter. But it's possible.

schoettl commented:
We could also add different quote character to the settings. But I'll leave that for future, don't need it currently.

zenhack commented:
Merged, thanks again!

schoettl commented:
Thanks!

zenhack commented:
Published as 1.1.0, hasn't shown up on package.elm-lang.org but I assume that's just a caching thing and it will update soon.

FYI, `elm bump` had some complaints about the docs, which I fixed:

- Some of the new exported identifiers weren't listed in `@docs`, so I added them.
- The `@docs` you added in front of the markdown footnote for the RFC is actually incorrect; you're only supposed to list identifiers there. It's fine as-is, and you can see on the package index that the RFC link works correctly.

zenhack commented:
Ack, another thing I missed: the `Settings` type was not exposed. Fixed and published as `1.2.0`.
